lib/components/primitive-components/Trace/get-obstacles-for-trace.ts

```diff
@@ -110,5 +152,51 @@ export const getSchematicObstaclesForTrace = (trace: Trace): Obstacle[] => {
     // }
   }
 
+  // Add schematic border as four thin rect obstacles (with padding)
+  if (isFinite(minX) && isFinite(maxX) && isFinite(minY) && isFinite(maxY)) {
+    const PADDING = 1
+    const left = minX - PADDING
+    const right = maxX + PADDING
+    const top = maxY + PADDING
+    const bottom = minY - PADDING
+    const thickness = 0.01
+    // Top border (horizontal)
+    obstacles.push({
+      type: "rect",
+      layers: ["top"],
+      center: { x: (left + right) / 2, y: top },
+      width: right - left,
+      height: thickness,
+      connectedTo: [],
+    })
+    // Bottom border (horizontal)
+    obstacles.push({
+      type: "rect",
+      layers: ["top"],
+      center: { x: (left + right) / 2, y: bottom },
+      width: right - left,
+      height: thickness,
+      connectedTo: [],
+    })
+    // Left border (vertical)
+    obstacles.push({
+      type: "rect",
+      layers: ["top"],
+      center: { x: left, y: (top + bottom) / 2 },
+      width: thickness,
+      height: top - bottom,
+      connectedTo: [],
+    })
+    // Right border (vertical)
+    obstacles.push({
+      type: "rect",
+      layers: ["top"],
+      center: { x: right, y: (top + bottom) / 2 },
+      width: thickness,
+      height: top - bottom,
+      connectedTo: [],
+    })
+  }
```

Comment:
Great comment, obstacle construction could also be isolated to a separate fn here, eg getObstaclesFromBounds(bounds,opts:{ padding })